### PR TITLE
Uses latest AM.Collection 7.x which uses AM 10.x

### LINKED ISSUE #36: Method not found: 'IMapper.Map(Object, Type, Type)'
Hi,
I did exactly like what the document says but I got this exception:

```
System.MissingMethodException: Method not found: 'System.Object AutoMapper.IMapper.Map(System.Object, System.Type, System.Type)'.
   at AutoMapper.EntityFrameworkCore.Persistence`1.GetEquivalenceExpression(Type type, Object from)
   at AutoMapper.EntityFrameworkCore.Persistence`1.InsertOrUpdateAsync(Type type, Object from, CancellationToken cancellationToken)
   at WebApplication1.Controllers.HomeController.Update(CategoryViewModel viewModel)
```

Startup.cs:
```c#
services.AddDbContext<DataContext>(builder => builder.UseSqlServer("Connection String"));

services.AddAutoMapper((provider, config) =>
{
   config.AddCollectionMappers();
   config.UseEntityFrameworkCoreModel<DataContext>(provider);
}, typeof(DataContext).Assembly);
```

Usage:
```c#
public class CategoryService
{
   public CategoryService(IMapper mapper) { } .......

   public Task Save(CategoryViewModel viewModel)
   {
      return _db.Categories.Persist(_mapper).InsertOrUpdateAsync(viewModel);
   }
}
```

Packages:
* AutoMapper 10.0.0
* AutoMapper.Collection 7.0.0
* AutoMapper.Collection.EntityFrameworkCore 1.0.1
* AutoMapper.Extensions.ExpressionMapping 4.0.0
* AutoMapper.Extensions.Microsoft.DependencyInjection 8.0.1
* Microsoft.EntityFrameworkCore 3.1.6

### PR DESCRIPTION
**Pull request by TylerCarlson1**
-Updating version to 7.0.0 to be consistent with AM.Collection's version
- Also update Entity framework to 3.1.10.  First before 5.0.0

@Tasteful Been a long time but is there any issues going forward with EF Core versions?  At least for now immediate future?

Fix #36 #45

### CODE REVIEW

File: src/AutoMapper.Collection.EntityFrameworkCore/AutoMapper.Collection.EntityFrameworkCore.csproj [hunk 1]
```diff
@@ -15,16 +15,16 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AutoMapper.Collection" Version="5.0.0" />
+    <PackageReference Include="AutoMapper.Collection" Version="7.0.0" />
```
Comment (Tasteful) [inline src/AutoMapper.Collection.EntityFrameworkCore/AutoMapper.Collection.EntityFrameworkCore.csproj:18]: Maybe this should be as below to make the reference only to work with the 7-version of AMC that (if I remember correctly is mapping to specific major version of AM.
```suggestion
    <PackageReference Include="AutoMapper.Collection" Version="[7.0.0,8.0.0)" />
```

**Review by Tasteful — APPROVED** [review]

**Comment by Tasteful** [comment]
Regarding to https://docs.microsoft.com/en-us/ef/core/what-is-new/ so is EFCore 2.1 the LTS release and supported to August 2021. Updating references here to 3 version will make the package to not work with all the LTS versions.

**Comment by TylerCarlson1** [comment]
I don't understand.  EFCore 3.1 is LTS as well, and upgrading will possibly cause breaking changes.

Is it so they leave at 2.1 and then the user can push to 3.1 or later if they want because it doesn't seem to break this code API wise?

**Comment by Tasteful** [comment]
Both the 2 and 3 major version have an LTS release each.

Using the 2 version in the reference make it possible for the user to upgrade to the 3 version. The other direction is not working so if AMC using 3 user cant downgrade to version 2.

So yes, to be compatible with most users, keep the 2.1 reference.

**Comment by risogolo** [comment]
Many thanks

**Comment by TylerCarlson1** [comment]
@risogolo Having issues with pushing to NuGet.com right now.  If you use AM Pre-Release on myget you'll be able to access 7.0.0

**Pull request merged**